### PR TITLE
build(docs): update throws and examples sections to new titles

### PR DESCRIPTION
**Pull request by RishhiB**
Consume the updated `api-markdown-documenter` package to allow custom titles for the `@example` and `@throws` sections.

1. The `api-markdown-documenter` package has utility functions `createExamplesSection` and `createThrowsSection` in `Helpers.ts` which was updated to include a optional headingTitle param in v0.9.0. These functions generate sections that describe the `@example` and `@throws` comments on an API item, respectively.
   
2. At present, the headings "Examples" and "Throws" are hard-coded for these sections. For FF.com, we aim to use custom titles "Usage" for the `@example` section and "Error Handling" for the `@throws` section.
   
3. Adjustments on FF.com:
   - The default title in the examples section should be changed from "Examples" to "Usage".
   - The default title in the throws section should be changed from "Throws" to "Error Handling".

image
image

[AB#5581](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/5581)

### CODE REVIEW

File: docs/api-markdown-documenter/api-documentation-layout.js [hunk 1]
```diff
@@ -84,7 +84,7 @@ function layoutContent(apiItem, itemSpecificContent, config) {
 	}
 
 	// Render examples (if any)
-	const renderedExamples = LayoutUtilities.createExamplesSection(apiItem, config);
+	const renderedExamples = LayoutUtilities.createExamplesSection(apiItem, config, "Usage");
```
Comment (scottn12) [inline docs/api-markdown-documenter/api-documentation-layout.js:87]: nit: maybe let's move out the string constants `"Usage"` and `"Error Handling"` out into consts at the top of this function so they are easier to find in the future?

File: docs/package.json [hunk 2]
```diff
@@ -39,7 +39,7 @@
 		"start": "hugo server"
 	},
 	"dependencies": {
-		"@fluid-tools/api-markdown-documenter": "^0.8.3",
+		"@fluid-tools/api-markdown-documenter": "^0.9.0",
```
Comment (scottn12) [inline docs/package.json:42]: What's the reasoning to bump this package in this PR?

**Review by scottn12 — APPROVED** [review]
Looks good! Left a tiny nit if you want :)

File: docs/package.json (hunk 2 above)
Comment (Josmithr) [reply docs/package.json:42]: The new version is needed in order to specify heading names to `createThrowsSection` and `createExamplesSection`.

Comment (Josmithr) [reply docs/package.json:42]: Probably worth calling this out in the PR description.

**Review by Josmithr — APPROVED** [review]
Looks good! I agree with Scott's comments, but otherwise this looks ready to go!

**Pull request merged**